Add Gen 2 Remix to snippets folder

bwreid opened the pull request:
## Description

Second attempt at adding Gen 2 Remix to our snippets folder, as requested here: [Jira Ticket](https://builder-io.atlassian.net/browse/EDU-152?atlOrigin=eyJpIjoiMjA4MzMxMzUwOTgyNDU1M2E4ZjVmOTYzYTdiNzY1MmIiLCJwIjoiaiJ9).

sidmohanty11 reviewed (commented):
hey @bwreid ! thanks for the PR

Can you please add `remix` to the CI here: https://github.com/BuilderIO/builder/blob/9458cc37faf16f0bf561fae5510f70bd7b4d033e/.github/workflows/ci.yml#L122-L138 so that it runs the snippet tests with this setup

sidmohanty11 commented on packages/sdks/snippets/remix/package-lock.json:
lock file is not required here as this is inside an nx monorepo, if you just do `yarn` it will reuse packages

bwreid replied:
Got it, thanks!

bwreid commented:
@sidmohanty11 Those updates have been made. Let me know if you have any other thoughts!

sidmohanty11 commented on packages/sdks/snippets/remix/app/routes/editable-region.tsx:
```diff
@@ -0,0 +1,59 @@
+      userAttributes: {
+        urlPath: window.location.pathname,
+      },
+      options: getBuilderSearchParams(new URL(location.href).searchParams),
```
```suggestion
```
this is not needed, check: https://builder-io.atlassian.net/browse/EDU-522

sidmohanty11 commented on packages/sdks/snippets/remix/app/routes/($slug)._index.tsx:
```diff
@@ -0,0 +1,46 @@
+  const urlPath = `/${params['slug'] || ''}`;
+
+  const page = await fetchOneEntry({
+    model: 'page',
```
nit: can you add a model name variable similar to how you've handled in all of the other pages and re-use it?

sidmohanty11 commented on packages/sdks/snippets/remix/app/routes/($slug)._index.tsx:
```diff
@@ -0,0 +1,46 @@
+
+  // Render the page content from Builder.io
+  return (
+    <Content model="page" apiKey={apiKey} content={page as BuilderContent} />
```
does it not auto infer the types?

sidmohanty11 commented on packages/sdks/snippets/remix/app/routes/($slug)._index.tsx:
```diff
@@ -0,0 +1,46 @@
+    apiKey: apiKey,
+    options: getBuilderSearchParams(url.searchParams),
+    userAttributes: { urlPath },
+    fetch: webFetch,
```
is this required? we can keep the snippet minimal and use this example in fetch options snippets

sidmohanty11 reviewed (commented):
@bwreid Looks great! Thank you very much for adding this! Left a few comments. Please remove the usage of `getBuilderSearchParams` everywhere in the codebase and we should be able to merge this!

bwreid commented:
@sidmohanty11 Thanks for the review! I updated the code as you suggested. I apologize for not seeing the ticket Uttej worked on about removing `getBuilderSearchParams()`.

I removed `getBuilderSearchParams()` from the Gen 2 Remix code, but not from the rest of the codebase, as Uttej is still working on that ticket. Please don't hesitate to let me know if you have any concerns about that, but I didn't want to get in the middle of what he was working on.

sidmohanty11 reviewed (approved):
🙌
